src/dhcp/aca_dhcp_server.cpp

```diff
@@ -0,0 +1,735 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_dhcp_server.h"
+#include "aca_log.h"
+#include "goalstateprovisioner.grpc.pb.h"
+#include <errno.h>
+#include <arpa/inet.h>
+#include "aca_ovs_control.h"
+#include "aca_ovs_l2_programmer.h"
+
+using namespace std;
+using namespace aca_dhcp_programming_if;
+
+namespace aca_dhcp_server
+{
+ACA_Dhcp_Server::ACA_Dhcp_Server()
+{
+  _init_dhcp_db();
+  _init_dhcp_msg_ops();
+  _init_dhcp_ofp();
+}
+
+ACA_Dhcp_Server::~ACA_Dhcp_Server()
+{
+  _deinit_dhcp_db();
+
+  _deinit_dhcp_ofp();
+}
+
+void ACA_Dhcp_Server::_init_dhcp_db()
+{
+  try {
+    _dhcp_db = new unordered_map<string, dhcp_entry_data>;
+  } catch (const bad_alloc &e) {
+    return;
+  }
+
+  _dhcp_entry_thresh = 0x10000; //10K
+}
+
+void ACA_Dhcp_Server::_deinit_dhcp_db()
+{
+  delete _dhcp_db;
+  _dhcp_db = nullptr;
+  _dhcp_entry_thresh = 0;
+}
+
+void ACA_Dhcp_Server::_init_dhcp_ofp()
+{
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer *pL2Prog = nullptr;
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  pL2Prog = &(aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance());
+
+  // adding dhcp default flows
+  pL2Prog->execute_openflow_command("add-flow br-int \"table=0,priority=25,udp,udp_src=68,udp_dst=67,actions=CONTROLLER\"",
+                                    not_care_culminative_time, overall_rc);
+  return;
+}
+
+void ACA_Dhcp_Server::_deinit_dhcp_ofp()
+{
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer *pL2Prog = nullptr;
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  pL2Prog = &(aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance());
+
+  // adding dhcp default flows
+  pL2Prog->execute_openflow_command("del-flow br-int \"udp,udp_src=68,udp_dst=67\"",
```

Comment:
You can just do this in one line, to be consistant with the rest of ACA code:
ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
cmd_string, not_care_culminative_time, overall_rc);